### PR TITLE
Action: AWS Delete EBS Snapshot

### PR DESCRIPTION
**Pull request by shloka-bhalgat-unskript**
### Testing

I don't have adequate permissions to delete a snapshot

Screenshot 2023-04-13 at 3 07 40 PM

Screenshot 2023-04-13 at 3 19 02 PM

### Checklist:
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] Any dependent changes have been merged and published. 

### Documentation
Make sure that you have documented corresponding changes in this repository.

### CODE REVIEW

**Review by dougsillars — DISMISSED** [review]

**Review by dougsillars — DISMISSED** [review]

File: AWS/legos/aws_delete_ebs_snapshot/aws_delete_ebs_snapshot.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    "action_title": "AWS Delete EBS Snapshot",
+    "action_description": "Delete EBS Snapshot for an EC2 instance",
+    "action_type": "LEGO_TYPE_AWS",
+    "action_entry_function": "aws_delete_ebs_snapshot",
+    "action_needs_credential": true,
+    "action_output_type": "ACTION_OUTPUT_TYPE_LIST",
+    "action_is_remediation": true,
```
Comment (amit-chandak-unskript) [inline AWS/legos/aws_delete_ebs_snapshot/aws_delete_ebs_snapshot.json:8]: what is is_remediation?

File: AWS/legos/aws_delete_ebs_snapshot/aws_delete_ebs_snapshot.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    "action_supports_iteration": true,
+    "action_supports_poll": true,
+    "action_categories": ["CATEGORY_TYPE_COST_OPT", "CATEGORY_TYPE_SRE","CATEGORY_TYPE_AWS","CATEGORY_TYPE_EC2"]
```
Comment (amit-chandak-unskript) [inline AWS/legos/aws_delete_ebs_snapshot/aws_delete_ebs_snapshot.json:11]: category should be EBS, not EC2

File: AWS/legos/aws_delete_ebs_snapshot/aws_delete_ebs_snapshot.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        description='EBS snapshot ID. Eg: "snap-34bt4bfjed9d"')
+
+
+def aws_delete_ebs_snapshots_printer(output):
+    if output is None:
+        return
+    pprint.pprint(output)
+
+
+def aws_delete_ebs_snapshots(handle, region: str, snapshot_id: str) -> Dict:
```
Comment (amit-chandak-unskript) [inline AWS/legos/aws_delete_ebs_snapshot/aws_delete_ebs_snapshot.py:25]: can we be careful with the name, this is only deleting one snapshot, right, so please match the name of the function to the directory and filename

File: AWS/legos/aws_delete_ebs_snapshot/aws_delete_ebs_snapshot.json (hunk 1 above)
Comment (shloka-bhalgat-unskript) [reply AWS/legos/aws_delete_ebs_snapshot/aws_delete_ebs_snapshot.json:8]: error. removed it.

File: AWS/legos/aws_delete_ebs_snapshot/aws_delete_ebs_snapshot.py (hunk 3 above)
Comment (shloka-bhalgat-unskript) [reply AWS/legos/aws_delete_ebs_snapshot/aws_delete_ebs_snapshot.py:25]: sure...

File: AWS/legos/aws_delete_ebs_snapshot/aws_delete_ebs_snapshot.json (hunk 1 above)
Comment (shloka-bhalgat-unskript) [reply AWS/legos/aws_delete_ebs_snapshot/aws_delete_ebs_snapshot.json:8]: So for all the remediation legos, we are adding `action_is_remediation` .

File: AWS/legos/aws_delete_ebs_snapshot/aws_delete_ebs_snapshot.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+    "action_title": "AWS Delete EBS Snapshot",
+    "action_description": "Delete EBS Snapshot for an EC2 instance",
+    "action_type": "LEGO_TYPE_AWS",
+    "action_entry_function": "aws_delete_ebs_snapshot",
+    "action_needs_credential": true,
+    "action_output_type": "ACTION_OUTPUT_TYPE_LIST",
+    "action_supports_iteration": true,
+    "action_supports_poll": true,
```
Comment (dougsillars) [inline AWS/legos/aws_delete_ebs_snapshot/aws_delete_ebs_snapshot.json:9]: should this have the "is remediation" line?

Comment (shloka-bhalgat-unskript) [reply AWS/legos/aws_delete_ebs_snapshot/aws_delete_ebs_snapshot.json:9]: @amit-chandak-unskript  said we don't need it... I've updated the PR.

Comment (dougsillars) [reply AWS/legos/aws_delete_ebs_snapshot/aws_delete_ebs_snapshot.json:9]: :)

**Review by dougsillars — APPROVED** [review]

**Review by amit-chandak-unskript — APPROVED** [review]

**Pull request merged**